### PR TITLE
Add Identity docs

### PR DESCRIPTION
**Pull request by eike-hass**
# Description of change

Add docs for Identity Rebased alpha version

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Documentation Fix

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [X] I have followed the contribution guidelines for this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/content/references/iota-identity/iota-did-method-spec.mdx [hunk 1]
```diff
@@ -0,0 +1,228 @@
+---
+title: IOTA DID Method Specification v1.0
```
Comment (eike-hass) [inline docs/content/references/iota-identity/iota-did-method-spec.mdx]:
```suggestion
title: IOTA DID Method Specification v2.0
```

File: docs/content/developer/iota-identity/getting-started/rust.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+identity_iota = { version = "1.4.0" }
+```
+
+### Development Release
+
+This version matches the `main` branch of this repository. It has all the **latest features**, but as such, it **may also have undocumented breaking changes**.
+
+```toml
+[dependencies]
+identity_iota = { git = "https://github.com/iotaledger/identity.rs", branch = "main"}
```
Comment (eike-hass) [inline docs/content/developer/iota-identity/getting-started/rust.mdx]:
```suggestion
identity_iota = { git = "https://github.com/iotaledger/identity.rs", tag = "v1.6.0-alpha"}
```

File: docs/content/developer/iota-identity/getting-started/rust.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+## Requirements
+
+- [Rust](https://www.rust-lang.org/) (>= 1.62)
+- [Cargo](https://doc.rust-lang.org/cargo/) (>= 1.62)
+
+## Include the Library
+
+To include IOTA Identity in your project, add it as a dependency in your `Cargo.toml`:
+
+### Latest Stable Release
```
Comment (eike-hass) [inline docs/content/developer/iota-identity/getting-started/rust.mdx]: We should delete the next sections until the actual include

**Comment by Dr-Electron** [comment]
Fixed some of the broken things. Just two things.

1) Let's not use `Markdown Extra` anchors. We decided to not use them because of the [mentioned reasons](https://github.com/iotaledger/iota/issues/2167)

2) Tags: They are now checked by docusaurus so we don't create similar tags and can only use the ones listed with a proper description and so on.

So if you think a tag is worth to be added, you have to add it to content/tags.yml

**Comment by Dr-Electron** [comment]
As discussed on slack I removed the tags

**Comment by Dr-Electron** [comment]
Btw you might want to add a sidebar so we can actually see the docs :trollface:

File: docs/content/sidebars/developer.js [hunk 4]
```diff
@@ -214,6 +214,98 @@ const developer = [
         label: 'NFT',
         items: ['developer/iota-101/nft/create-nft', 'developer/iota-101/nft/rent-nft'],
     },
+    {
```
Comment (lucas-tortora) [inline docs/content/sidebars/developer.js]: Please move this after the dev cheat sheet, it kind of gets lost where it is now

**Review by lucas-tortora — CHANGES_REQUESTED** [review]

File: docs/content/sidebars/developer.js (hunk 4 above)
Comment (lucas-tortora) [reply docs/content/sidebars/developer.js]: Once it's available on the testnet we could consider giving the docs their own section, as the ts sdk

File: docs/content/developer/iota-identity/explanations/decentralized-identifiers.mdx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+A DID is a unique identifier that can be resolved to a DID Document. This document contains data such as public keys, enabling the holder to prove ownership over their personal data, but also URIs that link to public information about the identity. DIDs are the fundamental building blocks of decentralized digital identity.
+This implementation complies with the [DID specifications v1.0](https://www.w3.org/TR/did-core//) from the World Wide Web Consortium (W3C).
+
+In the IOTA Identity framework, we have implemented the DID standard according to the `iota` [DID Method Specification](../../../references/iota-identity/iota-did-method-spec.mdx). Other implementations of DID on IOTA must follow the `iota` DID Method Specification if they want to use the `iota` method name. Libraries implementing the `iota` DID Method Specification are provided for [Rust](../getting-started/rust.mdx).
+
+An example of a DID conforming to the `iota` method specification:
+`did:iota:0xe4edef97da1257e83cbeb49159cfdd2da6ac971ac447f233f8439cf29376ebfe`
+
+## Decentralized Identifiers
```
Comment (Dr-Electron) [inline docs/content/developer/iota-identity/explanations/decentralized-identifiers.mdx]: Isn't this a duplicate?

**Review by Dr-Electron — APPROVED** [review]

File: docs/content/sidebars/developer.js (hunk 4 above)
Comment (eike-hass) [reply docs/content/sidebars/developer.js]: @lucas-tortora It is available on testnet

Comment (eike-hass) [reply docs/content/sidebars/developer.js]: where would you suggest to move it then?

Comment (lucas-tortora) [reply docs/content/sidebars/developer.js]: Let me put something together real quick

File: docs/content/developer/iota-identity/explanations/decentralized-identifiers.mdx (hunk 5 above)
Comment (eike-hass) [reply docs/content/developer/iota-identity/explanations/decentralized-identifiers.mdx]: true

File: docs/content/sidebars/developer.js (hunk 4 above)
Comment (lucas-tortora) [reply docs/content/sidebars/developer.js]: https://github.com/iotaledger/iota/pull/4381

File: docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+If you want to use the main IOTA network,
+you will need an an address with actual IOTA funds to create a new Identity.
+
+:::
+
+<div className={'hide-code-block-extras'}>
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+```rust reference
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx]: we changed the import strategy for the move docs, so these reference imports aren't working 

image

Comment (lucas-tortora) [reply docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx]: https://github.com/saucelabs/docusaurus-theme-github-codeblock

File: docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+https://github.com/iotaledger/identity.rs/blob/v1.6.0-alpha/examples/utils/utils.rs#L100
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx]: Line numbers seem to be wrong

Comment (lucas-tortora) [reply docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/blob/v1.6.0-alpha/examples/utils/utils.rs#L99
```

File: docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+the DID inside the document.
+
+:::
+
+<div className={'hide-code-block-extras'}>
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+```ts reference
+https://github.com/iotaledger/identity.rs/blob/v1.6.0-alpha/examples/utils/utils.rs#L38
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/blob/v1.6.0-alpha/examples/utils/utils.rs#L49-L58
```

File: docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+:::info Controller Capability
+When a new `Identity` is created the transaction's sender is assumed to be its controller and thus a `ControllerCap` is sent to its address.
+:::
+
+<div className={'hide-code-block-extras'}>
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+```rust reference
+https://github.com/iotaledger/identity.rs/blob/v1.6.0-alpha/examples/utils/utils.rs#L60
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/blob/v1.6.0-alpha/examples/utils/utils.rs#L60-L64
```

File: docs/content/developer/iota-identity/how-tos/decentralized-identifiers/delete.mdx [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+### Example
+
+The following example demonstrates how a controller destroys an IOTA Identity.
+
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+```rust reference
+https://github.com/iotaledger/identity.rs/blob/v1.6.0-alpha/examples/0_basic/4_delete_did.rs
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/decentralized-identifiers/delete.mdx]: this file doesn't exist

File: docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx (hunk 6 above)
Comment (eike-hass) [reply docs/content/developer/iota-identity/how-tos/decentralized-identifiers/create.mdx:42]: what is required to fix it?

File: docs/content/developer/iota-identity/how-tos/verifiable-credentials/create.mdx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+</div>
+
+### Validate a VC
+
+<div className={'hide-code-block-extras'}>
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+```rust reference
+https://github.com/iotaledger/identity.rs/blob/v1.6.0-alpha/examples/0_basic/5_create_vc.rs#L81-L89
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/verifiable-credentials/create.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/blob/v1.6.0-alpha/examples/0_basic/5_create_vc.rs#L80-L88
```

File: docs/content/developer/iota-identity/how-tos/verifiable-credentials/revocation.mdx [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+DLT's size constraints limit the size of the revocation list. With the assumption of only one such revocation list
+per the issuer's DID document, one may expect to be able to handle roughly 50k entries.
+
+:::
+
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+```rust reference
+https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/0_basic/7_revoke_vc.rs#L133
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/verifiable-credentials/revocation.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/0_basic/7_revoke_vc.rs#L132
```

File: docs/content/developer/iota-identity/how-tos/verifiable-credentials/revocation.mdx [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+entries in the status list, in contrast with `RevocationBitmap2022`, where the length of the list is limited by the DLT's constraints.
+
+Furthermore, `StatusList2021` introduces a new credential state: _suspended_. Suspended credentials are credentials that a validator will not accept as _valid_, but that might become valid again in the future. Instead, _revoked_ credentials **cannot** ever
+be valid again, as the _revoked_ state is irreversible.
+
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+```rust reference
+https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/1_advanced/8_status_list_2021.rs#L58-L62
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/verifiable-credentials/revocation.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/1_advanced/8_status_list_2021.rs#L57-L61
```

File: docs/content/developer/iota-identity/how-tos/verifiable-credentials/revocation.mdx [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+    "statusListIndex": "94567",
+    "statusListCredential": "https://example.com/credentials/status"
+}
+```
+
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+```rust reference
+https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/1_advanced/8_status_list_2021.rs#L145
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/verifiable-credentials/revocation.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/1_advanced/8_status_list_2021.rs#L144
```

File: docs/content/developer/iota-identity/how-tos/verifiable-credentials/revocation.mdx [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+
+However, this will also invalidate every VC signed with that verification method,
+meaning that the issuer will have to sign every VC with a different key to retain
+precise control over which credential is revoked.
+
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+```rust reference
+https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/0_basic/7_revoke_vc.rs#L158-L167
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/verifiable-credentials/revocation.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/0_basic/7_revoke_vc.rs#L159-L166
```

File: docs/content/developer/iota-identity/how-tos/verifiable-credentials/zero-knowledge-selective-disclosure.mdx [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+and send it back for verification. For this process the JWP in possession of the holder undergoes a transformation that allows the holder
+to conceal any fields from the credentials claims through the creation of a Zero Knowledge Proof (ZKP) of the issuer's signature and becomes a _presented JWP_.
+The proof value depends on the selected [JSON Proof Algorithm](https://www.ietf.org/archive/id/draft-ietf-jose-json-proof-algorithms-02.html) (JPA).
+
+<Tabs groupId="language" queryString>
+
+<TabItem value="rust" label="Rust">
+
+```rust reference
+https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/1_advanced/9_zkp.rs#L178-L204
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/verifiable-credentials/zero-knowledge-selective-disclosure.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/1_advanced/9_zkp.rs#L177-L203
```

File: docs/content/developer/iota-identity/how-tos/verifiable-credentials/zero-knowledge-selective-disclosure.mdx [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+
+The verifier decodes the received JPT presentation and asserts the validity of the ZKP it contains, thus proving the
+authenticity and integrity of the presented credential, without knowledge of any of the undisclosed fields and of the issuer signature.
+
+<Tabs groupId="language" queryString>
+
+<TabItem value="rust" label="Rust">
+
+```rust reference
+https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/1_advanced/9_zkp.rs#L225-L238
```
Comment (lucas-tortora) [inline docs/content/developer/iota-identity/how-tos/verifiable-credentials/zero-knowledge-selective-disclosure.mdx]:
```suggestion
https://github.com/iotaledger/identity.rs/tree/v1.6.0-alpha/examples/1_advanced/9_zkp.rs#L225-L237
```

**Review by lucas-tortora — CHANGES_REQUESTED** [review]

**Review by lucas-tortora — APPROVED** [review]

**Pull request merged**